src/Integration/Binding/CSharpVBProjectBinder.cs

```diff
@@ -156,10 +157,27 @@ private RuleSet FindDeclarationRuleSet(Project project, RuleSetDeclaration decla
             return ruleSetSerializer.LoadRuleSet(ruleSetFilePath);
         }
 
-        private bool ProjectHasAdditionalFile(Project project, string additionalFilePath)
+        public bool DoesProjectReferenceAdditionalFile(Project project, string solutionAdditionalFilePath)
         {
-            // ritag todo: 1. Check if there is another SonarLint.xml, 2. Check if the property is AdditionalFiles
-            return projectSystemHelper.IsFileInProject(project, additionalFilePath);
+            var additionalFile = projectSystemHelper.GetFileInProject(project, solutionAdditionalFilePath);
+
+            if (additionalFile == null)
```

Comment:
WIP: missing tests:
1. Project does not reference solution SonarLint.xml and has no other SonarLint.xml --> IsBindingRequired=true.
2. Project does not reference solution SonarLint.xml and has another SonarLint.xml --> Exception.
3. Project references solution SonarLint.xml but it's not AdditionalFiles --> IsBindingRequired=true.
4. Project references solution SonarLint.xml and it's AdditionalFiles --> IsBindingRequired=false.

I think it's also worth extracting it to another class as it starts to get too much (and potentially to a generic IsBindingRequiredIndicator, that could be used for ruleset as well, and this class will just be a composite of all the indicators).